Assembus/Assets/Scripts/MainScreen/Sidebar/HierarchyView/HierarchyItemController.cs

```diff
@@ -0,0 +1,633 @@
+        ///     The item on which the drag ended
+        /// </summary>
+        private static HierarchyItemController _dragItem;
+
+        /// <summary>
+        ///     The selected items before starting a drag
+        /// </summary>
+        private static List<HierarchyItemController> _selectedItems = new List<HierarchyItemController>();
+
+        /// <summary>
+        ///     The colors for the item
+        /// </summary>
+        public Color highlightedColor, normalColor;
+
+        /// <summary>
+        ///     The hierarchy view controller
+        /// </summary>
+        public HierarchyViewController hierarchyViewController;
+
+        /// <summary>
+        ///     The text view in which the name is shown
+        /// </summary>
+        public TextMeshProUGUI nameText;
+
+        /// <summary>
+        ///     The input field for renaming an item
+        /// </summary>
+        public TMP_InputField nameInput;
+
+        /// <summary>
+        ///     The matching game objects for the name label and input
+        /// </summary>
+        public GameObject nameTextObject, nameInputObject;
+
+        /// <summary>
+        ///     The rect transform of the item's content
+        /// </summary>
+        public RectTransform itemContent;
+
+        /// <summary>
+        ///     The expand button
+        /// </summary>
+        public GameObject expandButton, expandDown, expandRight;
+
+        /// <summary>
+        ///     The container of the item which contains all children
+        /// </summary>
+        public GameObject childrenContainer;
+
+        /// <summary>
+        ///     The context menu controller
+        /// </summary>
+        public ContextMenuController contextMenu;
+
+        /// <summary>
+        ///     The indicators for moving or putting an item
+        /// </summary>
+        public GameObject movingIndicator;
+
+        /// <summary>
+        ///     The background of the item
+        /// </summary>
+        public Image background;
+
+        /// <summary>
+        ///     The text for the preview of a list drag
+        /// </summary>
+        public TextMeshProUGUI dragPreviewText;
+
+        /// <summary>
+        ///     The preview object for a list drag
+        /// </summary>
+        public GameObject dragPreview;
+
+        /// <summary>
+        ///     The toast controller
+        /// </summary>
+        public ToastController toast;
+
+        /// <summary>
+        ///     The scroll view
+        /// </summary>
+        public ScrollRect scrollRect;
+
+        /// <summary>
+        ///     The item of the actual model
+        /// </summary>
+        [HideInInspector] public GameObject item;
+
+        /// <summary>
+        ///     The camera controller
+        /// </summary>
+        public CameraController cameraController;
+
+        /// <summary>
+        ///     The click detector instance
+        /// </summary>
+        public DoubleClickDetector clickDetector;
```

Comment:
Mayby naming the DoubleClickDetector doubleClickDetector because there alway can be a "SingleClickDetector"